spec/rspec/core/formatters/bisect_progress_formatter_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rspec/core/formatters/bisect_progress_formatter'
+
+module RSpec::Core::Formatters
+  RSpec.describe BisectProgressFormatter do
+    include FormatterSupport
+
+    def notify(*args)
+      reporter.publish(*args)
+    end
+
+    describe "dependency check results" do
+      it "confirms failures are dependent when the check passes" do
+        notify(
+          :bisect_dependency_check_finished,
+          :dependent_failures => true
+        )
+
+        expect(formatter_output.string).to match(/is order-dependent/)
+      end
+
+      it "states failures are independent when the check fails" do
+        notify(
+          :bisect_dependency_check_finished,
+          :dependent_failures => false
+        )
+
+        expect(formatter_output.string).to match(/is not order-dependent/)
+      end
+    end
+  end
+end
```

Comment:
This certainly works to cover the previously uncovered lines but I think I'd prefer to see them covered with a more integrated-style test (probably with the `FakeRunner`, though, for simplicity and faster speed).  I see a couple specific benefits of a more integrated style test over this approach:

* One of the big advantages I see from code coverage tools is that they can act as an automated dead code finder.  Consider if you test this by having an example in `coordinator_spec.rb` that causes these events to be exercised, and asserts on the total output, rather than testing them directly here.  In the future if we make changes that removes the need for these events, the code coverage will notify us these events are not covered and it'll serve it's purpose as a dead code finder.  OTOH, when you test them directly, they will _always_ be covered, even if we don't actually need or use them anymore.
* Here you've tested that when the formatter receives the event with `dependent_failures == false`, it outputs "is not order-dependent"...but as far as I can see, you have no test that demonstrates that the event is ever sent with that value.  By having a more integrated test you can easily cover both sides of the logic.

Obviously, I don't advocate always using a more integrated style (trade offs and all that) but it has some compelling advantages here, particular with the fact that we have the `FakeRunner`.  Thoughts?